src/libcore/slice/mod.rs

```diff
@@ -804,6 +804,61 @@ impl<T> [T] {
         (&self[..mid], &self[mid..])
     }
 
+    /// Divides one slice into two at an index.
+    ///
+    /// The first will contain all indices from `[0, mid)` (excluding
+    /// the index `mid` itself) and the second will contain all
+    /// indices from `[mid, len)` (excluding the index `len` itself).
+    ///
+    /// Returns `None` if `mid > len`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let v = [1, 2, 3, 4, 5, 6];
+    ///
+    /// {
+    ///     match v.split_at(0) {
+    ///         Some((left, right)) => {
+    ///             assert!(left == []);
+    ///             assert!(right == [1, 2, 3, 4, 5, 6]);
+    ///         },
+    ///         None => assert!(false),
+    ///     }
+    /// }
+    ///
+    /// {
+    ///     match v.split_at(2) {
+    ///         Some((left, right)) => {
+    ///             assert!(left == [1, 2]);
+    ///             assert!(right == [3, 4, 5, 6]);
+    ///         },
+    ///         None => assert!(false),
+    ///     }
+    /// }
+    ///
+    /// {
+    ///     match v.split_at(6) {
+    ///         Some((left, right)) => {
+    ///             assert!(left == [1, 2, 3, 4, 5, 6]);
+    ///             assert!(right == []);
+    ///         },
+    ///         None => assert!(false),
+    ///     }
+    /// }
+    ///
+    /// {
+    ///     assert!(v.split_at(7).is_none())
+    /// }
+    /// ```
+    #[unstable(feature = "try_split_at", issue = "54886")]
+    #[inline]
+    pub fn try_split_at(&self, mid: usize) -> Option<(&[T], &[T])> {
+        if mid > self.len() { None } else {
+            Some(unsafe { (self.get_unchecked(..mid), self.get_unchecked(mid..)) })
```

Comment:
Add a comment explaining why this is safe.

Reply:
More importantly, please justify using unsafe at all. While it's possible that something `if mid > self.len() { None } else { (&self[..mid], &self[mid..]) }` may end up with redundant bounds checks, in principle they can be optimized out and we should only use unsafe code if that optimization fails to fire.

Reply:
I just checked: it indeed elides the bounds checks and emits the same LLVM IR in either case.